### PR TITLE
🔧 Fix docutils deprecation in option parsing

### PR DESCRIPTION
**Pull request by agoose77**
This PR fixes the deprecation warnings raised by docutils' deprecation of `OptionParser`.

The help handling is a little ugly - we could introduce a switch on the docutils version like we do for the default settings. However, I *think* this approach will be less likely to break once the breaking change is introduced.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -118,6 +118,8 @@ ignore_missing_imports = true
 [tool.pytest.ini_options]
 filterwarnings = [
     "ignore:.*The default for the setting.*:FutureWarning",
+#    'ignore:.*The frontend\.OptionParser class will be replaced.*:DeprecationWarning',
+#    'ignore:.*The frontend\.Option class will be removed.*:DeprecationWarning'
```
Comment (agoose77) [inline pyproject.toml:122]:
```suggestion
```

**Comment by chrisjsewell** [comment]
Cheers @agoose77!

Note some suggestions on a fix were given here: https://github.com/executablebooks/MyST-Parser/issues/612#issuecomment-1265047994, maybe you want to double check that against your changes here?

**Comment by agoose77** [comment]
Thanks @chrisjsewell — I actually used that issue as a starting point. As such, I'm hopeful that this should fall in line with the recommended changes where possible.

**Review by chrisjsewell — APPROVED** [review]
Yep I think all good cheers

**Pull request merged**